scripts/utils/update_package.py

```diff
@@ -62,37 +86,109 @@ def get_sha256(url):
 # 1.2.3.4 -> 1.2.3
 # v1.2 -> 1.2
 # 1 -> 1
-def format_version(version):
-    match = re.match(r"v?(?P<version>\d+(.\d+){0,2})", version)
+def format_version(
+    version,
+):
+    match = re.match(
+        r"v?(?P<version>\d+(.\d+){0,2})",
+        version,
+    )
     if not match:
         raise ValueError(f"wrong version: {version}")
     return match.group("version")
 
 
-# Replace version in the package's nuspec file
-def update_nuspec_version(package, latest_version):
+# get content of nuspec file
+def get_nuspec(
+    package,
+):
     nuspec_path = f"packages/{package}/{package}.nuspec"
-    with open(nuspec_path, "r") as file:
+    with open(
+        nuspec_path,
+        "r",
+        encoding="utf-8",
+    ) as file:
         content = file.read()
-    latest_version, content = replace_version(latest_version, content)
-    with open(nuspec_path, "w") as file:
+    return content
```

Comment:
I think we should return the `nuspec_path` too (as we do in `get_install_script`). This way you do not need to redefine the `nuspec_path` in `update_dependencies` and `update_nuspec_version`.